core-rust/state-manager/src/state_manager.rs

```diff
@@ -425,21 +470,49 @@ where
-                        index: Some(index as u32),
-                        raw: raw_ledger_transaction,
-                        intent_hash: Some(intent_hash),
-                        notarized_transaction_hash: Some(notarized_transaction_hash),
-                        ledger_transaction_hash,
-                    });
-                    pending_transaction_results.push(PendingTransactionResult {
-                        intent_hash,
-                        notarized_transaction_hash,
-                        invalid_at_epoch,
-                        rejection_reason: None,
-                    });
+                Ok(processed_commit_result) => {
+                    match vertex_limits_tracker.check_post_execution(
+                        &self.vertex_limits_config,
+                        &processed_commit_result
+                            .local_receipt
+                            .local_execution
+                            .execution_metrics,
+                    ) {
+                        Ok(_) => {
+                            vertex_limits_tracker.advance(
+                                transaction_size,
+                                &processed_commit_result
+                                    .local_receipt
+                                    .local_execution
+                                    .execution_metrics,
+                            );
+                            duplicate_intent_hash_detector.record_committable_proposed(intent_hash);
+                            committable_transactions.push(CommittableTransaction {
+                                index: Some(index as u32),
+                                raw: raw_ledger_transaction,
+                                intent_hash: Some(intent_hash),
+                                notarized_transaction_hash: Some(notarized_transaction_hash),
+                                ledger_transaction_hash,
+                            });
+                            pending_transaction_results.push(PendingTransactionResult {
+                                intent_hash,
+                                notarized_transaction_hash,
+                                invalid_at_epoch,
+                                rejection_reason: None,
+                            });
+                        }
+                        Err(error) => {
+                            rejected_transactions.push(RejectedTransaction {
+                                index: index as u32,
+                                intent_hash: Some(intent_hash),
+                                notarized_transaction_hash: Some(notarized_transaction_hash),
+                                ledger_transaction_hash: Some(ledger_transaction_hash),
+                                error: format!("{:?}", &error),
+                            });
+                            // Note: we are not adding this transaction to [`pending_transaction_results`] because
```

Comment:
_(Note from author)_

We will want to revisit mempool eviction based on prepare failure as it's too agressive anyway.

Reply:
> We will want to revisit mempool eviction based on prepare failure as it's too aggressive anyway.

Maybe, but it's actually quite an effective performance boost in the 99% case where we have a linear vertex chain.

Reply:
Agree.
For this particular case though, would not be right to remove them (transaction just didn't fit in vertex limits).